zerocopy-derive/src/output_tests.rs

```diff
@@ -0,0 +1,193 @@
+// Copyright 2024 The Fuchsia Authors
+//
+// Licensed under a BSD-style license <LICENSE-BSD>, Apache License, Version 2.0
+// <LICENSE-APACHE or https://www.apache.org/licenses/LICENSE-2.0>, or the MIT
+// license <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your option.
+// This file may not be copied, modified, or distributed except according to
+// those terms.
+
+use synstructure::test_derive;
+
+macro_rules! impl_synstructure_derive {
+    ($($outer:ident => $inner:ident),* $(,)?) => {
+        $(
+            #[allow(non_snake_case)]
+            fn $outer(s: synstructure::Structure) -> proc_macro2::TokenStream {
+                super::$inner(s.ast())
+            }
+        )*
+    };
+}
+
+impl_synstructure_derive!(
+    KnownLayout => derive_known_layout_inner,
+    Immutable => derive_no_cell_inner,
+    TryFromBytes => derive_try_from_bytes_inner,
+    FromZeros => derive_from_zeros_inner,
+    FromBytes => derive_from_bytes_inner,
+    IntoBytes => derive_into_bytes_inner,
+    Unaligned => derive_unaligned_inner,
+);
+
+#[test]
+fn test_known_layout() {
+    test_derive! {
+        KnownLayout {
+            struct Foo;
```

Comment:
Let's exercise a bit more of the derive:
```suggestion
            struct Foo<const N: usize>
            where
                for<'a> Foo<N>: Sized;
```